pkg/clusterconfig/clusterconfig.go

```diff
@@ -102,21 +109,38 @@ func GetAWSConfig() (*Config, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	cfg.Storage.Type = StorageTypeS3
 	if installConfig.Platform.AWS != nil {
 		cfg.Storage.S3.Region = installConfig.Platform.AWS.Region
 	}
-	sec, err := client.Secrets(installerConfigNamespace).Get(installerAWSCredsName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("unable to read aws-creds secret: %v", err)
-	}
 
-	cfg.Storage.Type = StorageTypeS3
-	if v, ok := sec.Data["aws_access_key_id"]; ok {
-		cfg.Storage.S3.AccessKey = string(v)
-	}
-	if v, ok := sec.Data["aws_secret_access_key"]; ok {
-		cfg.Storage.S3.SecretKey = string(v)
+	// Look for a user defined secret to get the AWS credentials from first
+	sec, err := client.Secrets(imageRegistryNamespace).Get(userPrivateConfigName, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
+		glog.Infof("Optional user defined AWS credentials in secret \"%s/%s\" not found, ignoring.", imageRegistryNamespace, userPrivateConfigName)
+		// If no user defined secret is found, use the system one
+		sec, err = client.Secrets(installerConfigNamespace).Get(installerAWSCredsName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to get secret  \"%s/%s\": %v", installerConfigNamespace, installerAWSCredsName, err)
+		}
+		glog.Infof("Using cluster defined AWS credentials from secret \"%s/%s\"", installerConfigNamespace, installerAWSCredsName)
+		if v, ok := sec.Data["aws_access_key_id"]; ok {
```

Comment:
There should be a warning or an error if the key isn't in the secret.

Reply:
updated in next push